### PR TITLE
WIP: Support development-only source types

### PR DESCRIPTION
**Pull request by dccurtis**
Cost Management has local dev/setup modes that involve local source types to be stand-ins for AWS and Azure cloud providers.

I am working on enabling the team to run QE automation locally and this change would greatly help us in that effort.

If you are open to this; I would appreciate some assistance on what the best way to hide these types behind a development or debug flag so they are not present in CI or production.

### CODE REVIEW

**Comment by gmcculloug** [comment]
@gtanzillo If you have time would appreciate your thoughts on here.

**Comment by slemrmartin** [comment]
@dccurtis I'm opened to this. Can we also use these local SourceTypes (in topological inventory's collectors)? I suppose they are not paid. 

To hide it on CI/Prod: Look at https://github.com/RedHatInsights/e2e-deploy/pull/1537/files. This way you can set specific environment vars values for Prod etc. (https://github.com/RedHatInsights/e2e-deploy/tree/master/templates/sources/env)

Then you can skip some seeds (probably based on `-local` suffix) there:
https://github.com/RedHatInsights/sources-api/blob/0608f1d87e5022bed954e4d11f7b5b46f1b6579e/db/seeds.rb#L2 -> https://github.com/RedHatInsights/sources-api/blob/db8d0eeb5b5bd521face23bc0cc588741410070b/app/models/concerns/seeding_concern.rb#L7

**Comment by dccurtis** [comment]
Thanks for the tip @slemrmartin!  I'll get back to you when I have more. 👍

File: db/seeds/source_types.yml [hunk 1]
```diff
@@ -74,6 +74,81 @@ amazon:
         :initialValue: aws
   :vendor: Amazon
   :icon_url: "/apps/chrome/assets/images/partners-icons/aws.svg"
+amazon-local:
```
Comment (syncrou) [inline db/seeds/source_types.yml:77]: @dccurtis I agree with @slemrmartin in regards to not needing `-local` here if we use a `SUPPORTED_TYPES` pattern. Here are more examples around how that pattern was applied: https://github.com/RedHatInsights/topological_inventory-orchestrator/pull/79

Comment (dccurtis) [reply db/seeds/source_types.yml:77]: Thanks @syncrou !  I haven't forgotten about this.  Hope to have some updates to this branch this week.

**Comment by gmcculloug** [comment]
@eclarizio I fear `dippy-bot` might be stuck in a loop or something.

**Comment by gmcculloug** [comment]
@eclarizio I wonder if the bot looping on this PR has anything to do with it being a draft PR.  That's the only difference that jumps out at me.

**Comment by dccurtis** [comment]
Closing for now.  Haven't had time to get back to this lately.  Sorry for the bot noise!

**Comment by eclarizio** [comment]
@dccurtis No problem! Not your fault our bot went haywire haha

**Pull request closed without merging**